backend/service/topology/querybuilder.go

```diff
@@ -0,0 +1,181 @@
+package topology
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	sq "github.com/Masterminds/squirrel"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	topologyv1 "github.com/lyft/clutch/backend/api/topology/v1"
+)
+
+const (
+	column                   = "column"
+	columnIdentifer          = "column."
+	metadata                 = "metadata"
+	metadataIdentifer        = "metadata."
+	queryDefaultLimit uint64 = 100
+	maxResultLimit           = 1000
+)
+
+var metadataQueryValidator = regexp.MustCompile(`^[a-zA-Z0-9->'"]*$`)
```

Comment:
i think it would make more sense to check user input components rather than the overall query.

i can still include `'` and `--` in my query which lets me muck with the query (even though without spaces i'm not sure if it's possible to form a valid query)

Reply:
ah yeah true let me do that